src/compiler/xslt/invoke-compiled/group-invocation.xsl

```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet xmlns="http://www.w3.org/1999/XSL/TransformAlias"
+                xmlns:x="http://www.jenitennison.com/xslt/xspec"
+                xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                exclude-result-prefixes="#all"
+                version="3.0">
+
+   <!--
+      mode="x:group-invocation"
+      This mode groups the invocation instructions of compiled x:scenario into a single xsl:for-each.
+   -->
+
+   <xsl:template match="document-node()" as="element()*" mode="x:group-invocation">
+      <!-- xsl:call-template elements originating from x:scenario -->
+      <xsl:variable name="scenario-invokers" as="element(xsl:call-template)*"
+         select="xsl:call-template[processing-instruction(origin) eq 'scenario']" />
+
+      <!-- All the invocation instructions originating from x:scenario must be adjacent so that they
+         can be grouped in a single group. -->
+      <xsl:for-each select="
+            node()
+            [not(. intersect $scenario-invokers)]
+            [. >> $scenario-invokers[1]]
+            [$scenario-invokers[last()] >> .]">
+         <xsl:message terminate="yes">
+            <xsl:call-template name="x:prefix-diag-message">
+               <xsl:with-param name="message"
+                  select="'Unexpected node between invocation instructions of compiled scenario.'" />
```

Comment:
Do you know of a specific pattern that can trigger this error now, or is it more to make the implementation more robust or future-resistant as the compiler code evolves? Mostly just curious.

Reply:
Only for robustness. 02f0b2ebadb97b54928a9736b4bf61864d8baa10 clarifies it.